apps/frontend/src/pages/Profile.tsx

```diff
@@ -0,0 +1,264 @@
+    }
+  };
+
+  const handleImageUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target?.files && event.target?.files.length > 0) {
+      const file = event.target?.files[0];
+
+      if (file) {
+        const reader = new FileReader();
+
+        reader.onload = function (event) {
+          const base64String = event.target?.result as string;
+          setImageURL(base64String); // Base64 string of the file contents
+
+          if (!isImageValid(base64String)) {
+            alert("Invalid image type. Please upload a valid image file.");
+            return;
+          }
+
+          setImage(file);
+          setImageURL(URL.createObjectURL(file));
+          setEditMode(true);
+        };
+
+        reader.readAsDataURL(file); // This will trigger the onload function above
+      }
+    } else {
+      alert("No image selected.");
+    }
+
+    setIsImageUploaded(true);
+  };
+
+  const handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    field: string,
+  ) => {
+    setProfile({ ...profile, [field]: e.target.value });
+    setEditMode(true);
+  };
+
+  const handleSaveChanges = async () => {
+    let successImages: boolean | ImageURLObject = false;
+    if (isImageUploaded) {
+      try {
+        const listingImageBinary = imageURL.split(",")[1];
+        successImages = await asyncUploadSingleImage(listingImageBinary);
+      } catch (error) {
+        console.error("Error processing image:", error);
+      }
+    } else {
+      successImages = { url: profile.profile_picture };
+    }
+    if (successImages) {
+      profile.profile_picture = successImages.url;
+      try {
+        await _axios_instance.patch("/user", profile);
+        setOriginalProfile(profile);
+        console.log("Profile updated successfully");
+      } catch (error) {
+        console.error("Error saving changes:", error);
+        alert("Error saving changes. Please try again.");
+      }
+    } else {
+      alert("Error uploading picture. Please try again.");
+    }
+
+    setEditMode(false);
+    setIsImageUploaded(false);
+  };
+
+  const handleCancelChanges = () => {
+    setProfile(originalProfile);
+    setEditMode(false);
+  };
+
+  // TODO:
+  // Uncomment once Scott's PR is merged: https://github.com/UVicMartletplace/martletplace/pull/183/files
+  // Disable all text fields if !isCurrentUser
+  // Hide Save, Cancel, and Upload Profile buttons if !isCurrentUser
+  // Change "My Profile" to "<Name>'s Profile" if !isCurrentUser
+  // Check if id is the current user's id:
+  // const { user } = useUser();
+  // const [isCurrentUser, setIsCurrentUser] = useState(false);
+  // if (id === user.id) {
+  //   setIsCurrentUser(true);
+  // }
+
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "center",
+        alignItems: "center",
+        height: "100vh",
+      }}
+    >
+      {isDesktop && !isScreenSmall && (
```

Comment:
since !isScreenSmall already ensures the size is lg or bigger, the isDesktop check is redundant.
```suggestion
      {!isScreenSmall && (
```